### PR TITLE
fix: use platform dependent path for `join`, `resolve`, and `parse`

### LINKED ISSUE #2: Resolve behavior
If an absolute windows path is passed in `path.resolve` behavior, it will be simply joined

### PR DESCRIPTION
**Pull request by danielroe**
This takes advantage of the fact that win32 handles both `\` and `/` as valid separators and therefore can operate safely on a posix path.

1. resolves #2 by using `path.win32` (resolve)
1. handles unc paths by using `path.win32` (join)
1. resolves a parse issue where the drive letter was being stripped on windows
1. normalizes `toNamespacedPath` as well, for consistency

**An observation**: this entire library works (passes tests) in reverse using `path.win32` and normalising the result (as with 1 & 2 above)

With this PR merged, #3 should be passing.

### CODE REVIEW

**Comment by pi0** [comment]
Comparing [win32 implementation](https://github.com/nodejs/node/blob/8d83c470291d05f22010662380633bbddf0e7dcb/lib/path.js#L158) of `resolve` to the [POSIX one](https://github.com/nodejs/node/blob/8d83c470291d05f22010662380633bbddf0e7dcb/lib/path.js#L1091) there seems a huge difference (and performance impact). In the meantime:

- I think we can use platform dependant behavior for `resolve`/`join`, hoping no one cross runs windows paths in posix env
- In the longer term, either inline `posix.resolve` implementation to apply fixes or directly wrap. Since we do not need ALL windows features in a cross-env implementation.

File: src/index.ts [hunk 1]
```diff
@@ -51,13 +52,14 @@ export const basename: typeof path.basename = function (p, ext) {
...
+// TODO
 export const parse: typeof path.parse = function (p) {
-  return path.posix.parse(normalizeWindowsPath(p))
+  return path.parse(normalizeWindowsPath(p))
 }
 
 // toNamespacedPath
 export const toNamespacedPath: typeof path.toNamespacedPath = function (p) {
-  return p
+  return normalizeWindowsPath(p)
```
Comment (pi0) [inline src/index.ts:62]: This is also a platform-dependent API that POSIX behavior is to return input as-is....

**Comment by pi0** [comment]
I had to revert b90fdafda9c0d96f1238c4e72d80821e2a1bc3ab since seems added edge cases went more :( Let's try when merging test suite for better solution.
